cpp/src/join/conditional_join_kernels.cuh

```diff
@@ -67,20 +67,21 @@ __global__ void compute_conditional_join_output_size(
     &intermediate_storage[threadIdx.x * device_expression_data.num_intermediates];
 
   std::size_t thread_counter{0};
-  cudf::size_type const start_idx      = threadIdx.x + blockIdx.x * block_size;
-  cudf::size_type const stride         = block_size * gridDim.x;
-  cudf::size_type const left_num_rows  = left_table.num_rows();
-  cudf::size_type const right_num_rows = right_table.num_rows();
-  auto const outer_num_rows            = (swap_tables ? right_num_rows : left_num_rows);
-  auto const inner_num_rows            = (swap_tables ? left_num_rows : right_num_rows);
+  auto const start_idx                         = cudf::detail::grid_1d::global_thread_id();
+  cudf::thread_index_type const stride         = thread_index_type{block_size} * gridDim.x;
```

Comment:
Are we sure that `block_size` can be replaced with `blockDim.x`? If so, let's wait for #13996 and use `cudf::detail::grid_1d::grid_stride()`. If not, the use of `global_thread_id` isn't valid either.